### ISSUE TITLE
Confirm there is no slowdown

### ISSUE THREAD

**Issue by nrnrnr** [ndaniels/MRFy#9]
Confirm that commit 049883fd884e57679204104cacabd0f329051230 is at worst no slower than 2f9bd4fbc20ef50702bed2a01059f04bcdf03b44 on any machine or benchmark we know about.

(The first is "new HO Viterbi with strict scoring and `preceders` as a function".)

(The second is the ICFP version on the master branch.)

**Comment by BurntSushi** [comment]
Data (also in `mrfy-paper/EXPERIMENTS`).

Regardless of name, the first branch in each set is the first iteration of `hoViterbi`. The second branch is `hoViterbi` with strict scoring and `preceders` as a function. The third branch is ICFP.

```
Comparisons on many machines related to MRFy#9
==============================================

### 2013-05-15 21:13  Running short: three-node-record-orig-func-strict master three-node-record
...ode-record 12186bc     10.61s  x86_64  GHC 7.6.3  Core i7-3770 @ 3.40GHz
...unc-strict 049883f     10.47s  x86_64  GHC 7.6.3  Core i7-3770 @ 3.40GHz
master 2f9bd4f             9.05s  x86_64  GHC 7.6.3  Core i7-3770 @ 3.40GHz

### 2013-05-15 21:14  Running med: three-node-record-orig-func-strict master three-node-record
...ode-record 12186bc     55.70s  x86_64  GHC 7.6.3  Core i7-3770 @ 3.40GHz
...unc-strict 049883f     54.71s  x86_64  GHC 7.6.3  Core i7-3770 @ 3.40GHz
master 2f9bd4f            54.35s  x86_64  GHC 7.6.3  Core i7-3770 @ 3.40GHz

### 2013-05-15 21:01  Running short: three-node-record three-node-record-orig-func-strict master
...ode-record 12186bc     19.89s  x86_64  GHC 7.6.3  AMD Phenom II X6 1075T Processor
...unc-strict 049883f     19.42s  x86_64  GHC 7.6.3  AMD Phenom II X6 1075T Processor
master 2f9bd4f            17.14s  x86_64  GHC 7.6.3  AMD Phenom II X6 1075T Processor

### 2013-05-15 21:09  Running med: three-node-record three-node-record-orig-func-strict master
...ode-record 12186bc    104.15s  x86_64  GHC 7.6.3  AMD Phenom II X6 1075T Processor
...unc-strict 049883f    102.17s  x86_64  GHC 7.6.3  AMD Phenom II X6 1075T Processor
master 2f9bd4f           103.22s  x86_64  GHC 7.6.3  AMD Phenom II X6 1075T Processor

### 2013-05-15 21:51  Running short: three-node-record three-node-record-orig-func-strict master
...ode-record 12186bc     12.31s  x86_64  GHC 7.6.3  Core i7-3930K @ 3.20GHz
...unc-strict 049883f     11.10s  x86_64  GHC 7.6.3  Core i7-3930K @ 3.20GHz
master 2f9bd4f             9.62s  x86_64  GHC 7.6.3  Core i7-3930K @ 3.20GHz

### 2013-05-15 22:01  Running med: three-node-record three-node-record-orig-func-strict master
...ode-record 12186bc     63.14s  x86_64  GHC 7.6.3  Core i7-3930K @ 3.20GHz
...unc-strict 049883f     56.55s  x86_64  GHC 7.6.3  Core i7-3930K @ 3.20GHz
master 2f9bd4f            55.08s  x86_64  GHC 7.6.3  Core i7-3930K @ 3.20GHz


NR's benchmarks (branches are different names but correspond to above)
----------------------------------------------------------------------

### 2013-05-15 20:34  Running short: lazy-score icfp-branch mrfy v3-orig
v3-orig 12186bc           10.02s    i686  GHC 7.4.1  Core i5-3570 @ 3.40GHz
mrfy 049883f               8.59s    i686  GHC 7.4.1  Core i5-3570 @ 3.40GHz
icfp-branch 2f9bd4f        9.25s    i686  GHC 7.4.1  Core i5-3570 @ 3.40GHz

### 2013-05-15 20:42  Running short: icfp-branch lazy-score mrfy v3-orig
v3-orig 12186bc           20.11s    i686  GHC 7.4.1  AMD Phenom 9850
mrfy 049883f              19.10s    i686  GHC 7.4.1  AMD Phenom 9850 
icfp-branch 2f9bd4f       19.11s    i686  GHC 7.4.1  AMD Phenom 9850 

### 2013-05-15 20:45  Running med: icfp-branch lazy-score mrfy v3-orig
v3-orig 12186bc          113.64s    i686  GHC 7.4.1  AMD Phenom 9850
mrfy 049883f             107.21s    i686  GHC 7.4.1  AMD Phenom 9850
icfp-branch 2f9bd4f      123.53s    i686  GHC 7.4.1  AMD Phenom 9850
```

TL;DR - It looks like there is a small constant? slow down on my machines but not on @nrnrnr. My machines are all 64 bit with GHC 7.6.3 while @nrnrnr's are 32 bit with GHC 7.4.1.
